Docs/CERTIFICATE_GENERATION.md

```diff
@@ -0,0 +1,137 @@
+# Certificate Generation
+
+Use the script below to generate certificates for running an extension. 
+The certificate for the extension needs to be in the same chain as the 
+root certificate for the OpenSearch cluster defined in the `plugins.security.ssl.http.pemtrustedcas_filepath`
+setting of `opensearch.yml`
+
+
+## Certificate Generation Script
```

Comment:
@cwperks TLS solves the problem of identifying the extension. 
Shouldn't these scripts be part of the OpenSearch distribution packaged from security repo[1]?
These certs should be generated by a CA which security plugin understands (a.k.a security plugin tools should issue client certificates). 

[1] https://github.com/opensearch-project/security/tree/main/tools

Reply:
I included this in here for testing instructions to show how to test this change manually. Currently we have no tool for certificate generation, only documentation: https://opensearch.org/docs/latest/security/configuration/generate-certificates/

I don't think we should create a tool for certificate generation, but we should update the `install_demo_configuration.sh` tool to account for extensions. If we create a demo configuration with AD as an extension, then AD will need its own demo certs just like the node does in the demo configuration.

Reply:
That makes sense, thank you.